snapshot.go

```diff
@@ -0,0 +1,359 @@
+package frostdb
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/apache/arrow/go/v10/arrow/ipc"
+	"github.com/go-kit/log/level"
+	"github.com/google/btree"
+	"github.com/segmentio/parquet-go"
+
+	"github.com/polarsignals/frostdb/dynparquet"
+	schemav2pb "github.com/polarsignals/frostdb/gen/proto/go/frostdb/schema/v1alpha2"
+	snapshotpb "github.com/polarsignals/frostdb/gen/proto/go/frostdb/snapshot/v1alpha1"
+	"github.com/polarsignals/frostdb/parts"
+)
+
+const (
+	snapshotMagic = "FDBS"
+	dirPerms      = os.FileMode(0755)
+	filePerms     = os.FileMode(0640)
+)
+
+// segmentName returns a 20-byte textual representation of a snapshot file name
+// at a given txn used for lexical ordering.
+func snapshotFileName(tx uint64) string {
+	return fmt.Sprintf("%020d.fdbs", tx)
```

Comment:
Couldn't we use ULIDs for this too? Not sure about the intent here.

Reply:
This encodes the transaction into the filename, similar to how the WAL code does it. Is there a good reason to use ULIDs instead of this?

Reply:
Not really, makes sense. Let's leave it as is.